### PR TITLE
[Keyboard] Add Arrow Mechanics WINGS

### PR DESCRIPTION
**Pull request by arrowmechanics**
## Description

Add base support for Arrow Mechanics WINGS keyboard with all the essential features

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/arrowmechanics/wings/info.json [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+    "maintainer": "arrowmechanics",
+    "bootloader": "rp2040",
+    "diode_direction": "ROW2COL",
+    "encoder": {
+        "rotary": [
+            {"pin_a": "GP6", "pin_b": "GP7"},
+            {"pin_a": "GP6", "pin_b": "GP7"}
+        ]
+    },
+    "matrix_pins": {
```
Comment (zvecr) [inline keyboards/arrowmechanics/wings/info.json:13]:
```suggestion
    "features": {
        "bootmagic": true,
        "extrakey": true,
        "mousekey": true,
        "audio": true,
        "encoder": true,
        "rgb_matrix": true
    },
    "matrix_pins": {
```

Comment (arrowmechanics) [reply keyboards/arrowmechanics/wings/info.json:13]: applied, but we don't want to enable bootmagic by default, hence set to false

**Review by arrowmechanics — COMMENTED** [review]
applied review suggestions, please see one comment

File: keyboards/arrowmechanics/wings/info.json (hunk 1 above)
Comment (zvecr) [reply keyboards/arrowmechanics/wings/info.json:21]: We prefer it to be enabled, as it keeps things consistent and helps when there are issues and a user needs to be told how to perform an eeprom reset.

Comment (arrowmechanics) [reply keyboards/arrowmechanics/wings/info.json:21]: allright, keeping it then

**Review by arrowmechanics — COMMENTED** [review]
applied all review suggestions

**Review by waffle87 — APPROVED** [review]

**Comment by arrowmechanics** [comment]
No more functional changes in this PR! (just fixes if needed)

**Comment by arrowmechanics** [comment]
With following data-driven definition of a split keyboard having two encoders, one per hand, linter throws error with their indices, though everything compiles and works correctly:
```python
{
    "encoder": {
          "rotary": [
              {"pin_a": "GP6", "pin_b": "GP7", "resolution": 2}
          ]
    },
    "split": {
            "enabled": true,
            "transport": {
                "sync": {
                    "layer_state": true,
                    "matrix_state": true
                }
          }
    },
    "layouts": {
            "LAYOUT": {
                "layout": [
                    {"matrix": [1, 8], "x": 8.25, "y": 1.1507, "encoder": 0},
                    {"matrix": [7, 0], "x": 9.45, "y": 1.1507, "encoder": 1}
                ]
          }
    }

  ...
}
``` 

The hotfix is simply removing the other encoder from the matrix definition, as it is practically not needed (no LED is assigned to any of the encoders). In general, however, this is suspicious (issue report maybe?) as it seems to be an inconsistency between linter and the data-driven config requirements.

File: keyboards/arrowmechanics/wings/config.h [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#define SPLIT_HAND_PIN_LOW_IS_LEFT
```
Comment (zvecr) [inline keyboards/arrowmechanics/wings/config.h:19]: SPLIT_HAND_PIN is not defined (nor configured in info.json)
```suggestion
```

File: keyboards/arrowmechanics/wings/config.h [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+#define SERIAL_USART_FULL_DUPLEX
+#define MASTER_LEFT
```
Comment (zvecr) [inline keyboards/arrowmechanics/wings/config.h:21]:
```suggestion
```

**Review by zvecr — CHANGES_REQUESTED** [review]

**Comment by arrowmechanics** [comment]
I see, I'll stop and wait until You're ready :)14 mar 2024 12:06 Joel Challis ***@***.***> napisał(a):
image.png (view on web)
All the constant merging is doing is burning CI minutes and making other PRs workflow runs take longer.

**Comment by arrowmechanics** [comment]
Is there anything else that stops us from merging?

**Comment by arrowmechanics** [comment]
> 
> All the constant merging is doing is burning CI minutes and making other PRs workflow runs take longer.

Also, this might be a thing for thought when it comes to maintenance rules of the QMK repo - how should I know as a minor contributor, how many updates to the main branch should I expect before my PR is taken care of?

I understand my mistake now, but it's not that obvious in general for there are no releases, no time-related or anyhow strict rules of merging to main. It actually looks confusing as multiple minor updates were merged in the meantime without even passing the pipelines - maybe it's a good idea to elaborate on that in the docs?

**Comment by tzarc** [comment]
It's not obvious, because the vast majority of people aren't so eager to update their repo.
I run QMK CI locally, so I pay for electricity costs associated with it -- _normal_ use I'm okay with.

I've disabled CI for your PR for now. If/when collabs get back to it, they can remove the label and retrigger CI on demand.

**Comment by arrowmechanics** [comment]
> It's not obvious, because the vast majority of people aren't so eager to update their repo. I run QMK CI locally, so I pay for electricity costs associated with it -- _normal_ use I'm okay with.
> 
> I've disabled CI for your PR for now. If/when collabs get back to it, they can remove the label and retrigger CI on demand.

I really am keen, we're yet another start-up that took up QMK as a key element. Still I didn't know You run it locally.
This makes me think it might be good time to start doing this:
image
I promise to go for more as soon as we're profitable :P

File: keyboards/arrowmechanics/wings/readme.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+94-key ErgoDox-inspired SPLIT keyboard by [Arrow Mechanics](https://www.arrowmechanics.com/)
+
+* Keyboard Maintainer: [arrowmechanics](https://github.com/arrowmechanics/)
+* Hardware Supported: [Arrow Mechanics WINGS Keyboard set](https://www.arrowmechanics.com/produkty?view=product&id=8)
+* Hardware Availability: [Official Arrow Mechanics E-SHOP](https://www.arrowmechanics.com/produkty?view=product&id=8)
+
+Flashing example for this keyboard:
+```sh
+qmk flash -kb arrowmechanics/wings -km default
+```
```
Comment (drashna) [inline keyboards/arrowmechanics/wings/readme.md:14]:
```suggestion

    qmk flash -kb arrowmechanics/wings -km default

```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/arrowmechanics/wings/config.h (hunk 3 above)
[the remark this answers is not in the document]
Comment (arrowmechanics) [reply keyboards/arrowmechanics/wings/config.h]: done already 😄

**Comment by arrowmechanics** [comment]
This PR seems to have too many comments/issues vs. few changes that it actually introduces. Closing to create a clean version since it may take too much time to dig into things that are already fixed.

**Pull request closed without merging**